src/java.base/share/classes/jdk/internal/foreign/layout/AbstractLayout.java

```diff
@@ -53,8 +53,9 @@ public final L withName(String name) {
         return dup(byteAlignment(), Optional.of(name));
     }
 
+    @SuppressWarnings("unchecked")
     public final L withoutName() {
-        return dup(byteAlignment(), Optional.empty());
+        return name.isPresent() ? dup(byteAlignment(), Optional.empty()) : (L) this;
```

Comment:
Made this idempotent to make the `withoutName()` call in `makeSegmentViewVarHandle` cheaper.